src/index.ts

```diff
@@ -602,6 +602,14 @@ class ThreadPool {
     this._ensureMinimumWorkers()
     this.startingUp = false
   }
+  _ensureEnoughWorkersForTaskQueue(): void {
+    while (
+      this.workers.size < this.taskQueue.size &&
+      this.workers.size < this.options.maxThreads
+    ) {
+      this._addNewWorker()
+    }
+  }
```

Comment:
@overlookmotel Do you think this solution is an optimal one?

Reply:
@overlookmotel Let me know if you're there.

Reply:
I'm really sorry but I can't help more. As I mentioned on #42, I am very tight for time. The complexity (which we're discussing here) was exactly why I didn't feel I could make a PR myself. I find piscina's code pretty hard to disentangle.

But my guess is that your solution above is not quite right, and will create new workers when none is actually needed in some case.

My suggestion would be to look at the existing code for how it's handled when a thread is terminated due to use of an `AbortController`. That results in same situation (worker removed from pool, do we need to add a new one?) so I imagine the code for that contains a good solution.

Reply:
sure! ok, thank you so much! I'll investigate it.

Reply:
In AbortController the ensureMin approach is used which is not proper for our case!
I think let's stick with the `_ensureEnoughWorkersForTaskQueue` approach for now and let's see how it goes! Thanks for yout help.